stoix/utils/loss.py

```diff
@@ -105,46 +105,52 @@ def categorical_double_q_learning(
 
 
 def n_step_categorical_double_q_learning(
```

Comment:
Hey, I've added the n-step loss for C51. It seems to run correctly and the loss decreases steadily during training even though it doesn't perform as well as noisy DQN after a million timesteps. I would assume that learning to model the 3-step return distribution is a harder task that could require more time so I'm not surprised. What do you think?

Note: I had to reduce the buffer size to 50_000 (default: 500_000) and the batch size to 128 (default: 256) because of memory issues, which could also explain the slower learning.

```
Env: snake 6x6

TRAINER - Q loss: 0.697
EVALUATOR - Steps per second: 162082.629 
Episode length mean: 4000.000 
Episode length std: 0.000 
Episode length min: 4000.000 
Episode length max: 4000.000 
Episode return mean: 0.977 
Episode return std: 1.902 
Episode return min: 0.000 
Episode return max: 10.000

ABSOLUTE - Steps per second: 536091.367 
Episode length mean: 3898.128 
Episode length std: 626.234 
Episode length min: 13.000 
Episode length max: 4000.000 
Episode return mean: 0.914 
Episode return std: 1.617 
Episode return min: 0.000 
Episode return max: 9.000
```

Reply:
hmm this doesnt seem to be learning. Let me investigate